multi-transfer-esdt/src/transfer_tokens.rs

```diff
@@ -77,23 +78,35 @@ pub trait TransferTokensModule:
         payments: PaymentsVec<Self::Api>,
         all_token_data: ManagedVec<StolenFromFrameworkEsdtTokenData<Self::Api>>,
     ) -> PaymentsVec<Self::Api> {
+        let own_sc_address = self.blockchain().get_sc_address();
         let mut output_payments = PaymentsVec::new();
         for (payment, token_data) in payments.iter().zip(all_token_data.iter()) {
+            let token_balance = self.blockchain().get_esdt_balance(
+                &own_sc_address,
+                &payment.token_identifier,
+                payment.token_nonce,
+            );
+            if token_balance >= payment.amount {
```

Comment:
If there are 2 payments with amount X and Y for the same token_id, then you save the 2 payments in `output_payments`, and
`X < token_balance`
`Y < token_balance`
but `X + Y > token_balance`.

Reply:
Fair, but how do I solve this easily?